connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -2212,6 +2212,13 @@ DistributedHerderRequest addRequest(Callable<Void> action, Callback<Void> callba
     DistributedHerderRequest addRequest(long delayMs, Callable<Void> action, Callback<Void> callback) {
         DistributedHerderRequest req = new DistributedHerderRequest(time.milliseconds() + delayMs, requestSeqNum.incrementAndGet(), action, callback);
         requests.add(req);
+        // We don't need to synchronize here
+        // If the condition evaluates to true, we can and should trigger a wakeup
+        // If it evaluates to false because our request has suddenly been popped off of the queue, then
+        //     the herder is already running the request and no wakeup is necessary
```

Comment:
nit: unnecessary indentation?

Reply:
Was hoping to make the separation between the different conditions clearer (sort of but not quite like how we indent if/else chains in Java). Guess that doesn't read well; I've removed the extra indentation here.